### PR TITLE
Docker build fixes

### LINKED ISSUE #1402: Building withing Docker fails due to Git error
**Bug description:**
Trying to build within the Docker contains fails with a Git error:

```
fatal: Not a git repository (or any of the parent directories): .git
Error: Command failed: git rev-parse HEAD
fatal: Not a git repository (or any of the parent directories): .git

    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:652:15)
    at Object.<anonymous> (/usr/src/ferdi/node_modules/preval-build-info/scripts/build.js:63:22)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  status: 128,
  signal: null,
  output: [
    null,
    <Buffer >,
    <Buffer 66 61 74 61 6c 3a 20 4e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
  ],
  pid: 2380,
  stdout: <Buffer >,
  stderr: <Buffer 66 61 74 61 6c 3a 20 4e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
}
fatal: Not a git repository (or any of the parent directories): .git
Error: Command failed: git rev-parse --short HEAD
fatal: Not a git repository (or any of the parent directories): .git

    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:652:15)
    at Object.<anonymous> (/usr/src/ferdi/node_modules/preval-build-info/scripts/build.js:71:28)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  status: 128,
  signal: null,
  output: [
    null,
    <Buffer >,
    <Buffer 66 61 74 61 6c 3a 20 4e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
  ],
  pid: 2382,
  stdout: <Buffer >,
  stderr: <Buffer 66 61 74 61 6c 3a 20 4e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
}
fatal: Not a git repository (or any of the parent directories): .git
Error: Command failed: git describe --always --tag --abbrev=0
fatal: Not a git repository (or any of the parent directories): .git

    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:652:15)
    at Object.<anonymous> (/usr/src/ferdi/node_modules/preval-build-info/scripts/build.js:81:21)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  status: 128,
  signal: null,
  output: [
    null,
    <Buffer >,
    <Buffer 66 61 74 61 6c 3a 20 4e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
  ],
  pid: 2384,
  stdout: <Buffer >,
  stderr: <Buffer 66 61 74 61 6c 3a 20 4e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
}
fatal: Not a git repository (or any of the parent directories): .git
Error: Command failed: git rev-parse --abbrev-ref HEAD
fatal: Not a git repository (or any of the parent directories): .git

    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:652:15)
    at Object.<anonymous> (/usr/src/ferdi/node_modules/preval-build-info/scripts/build.js:91:24)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  status: 128,
  signal: null,
  output: [
    null,
    <Buffer >,
    <Buffer 66 61 74 61 6c 3a 20 4e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
  ],
  pid: 2386,
  stdout: <Buffer >,
  stderr: <Buffer 66 61 74 61 6c 3a 20 4e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
}
```

It seems we're trying to read the current commit hash from within the container, but we can't access the `.git` folder, since it is Dockerignored. As sending `.git` over to the Docker daemon is frowned upon, we should probably pass the commit hash with some other method to the Docker builder instead.

**Steps to reproduce:**
1. `docker build -t ferdi-package .`

### PR DESCRIPTION
**Pull request by vraravam**
### Description
Fixes issue with building inside of the docker container. The caveat of this is that the build information will not be available.

### Motivation and Context
Fixes #1402 

### Checklist:

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`$ npm run lint`)
- [x] I tested/previewed my changes locally

### CODE REVIEW

**Comment by vraravam** [comment]
@kris7t - there is 1 final error that I am unable to solve. could you please check if you have bandwidth?

**Comment by kris7t** [comment]
I tried after merging this and pruning local Docker containers & images, but it still displays the same error for me as before: https://linx.marussy.com/s/y6m1hwas.txt

**Comment by vraravam** [comment]
I am not sure if you are running the latest code. The above `not a git repo` issue should have been fixed with my commit - specifically this line:
`ENV PREVAL_BUILD_INFO_PLACEHOLDERS=true`

**Comment by vraravam** [comment]
Screenshot 1:
Screen Shot 2021-05-14 at 5 02 37 AM

Screenshot 2: 
Screen Shot 2021-05-14 at 5 02 37 AM

This later error is where I'm stuck now.

File: Dockerfile [hunk 1]
```diff
@@ -11,12 +11,16 @@ WORKDIR /usr/src/ferdi
...
-RUN npm i gulp@^4.0.0
-RUN npx lerna bootstrap
+RUN npm i node-gyp@8.0.0 node-sass@5.0.0 \
+    && npx lerna bootstrap
 
 COPY . .
 
-RUN npm run build
+# Note: This is being set to bypass the error with missing git repo information for the 'preval-build-info' module
+ENV PREVAL_BUILD_INFO_PLACEHOLDERS=true
```
Comment (vraravam) [inline Dockerfile:20]: @kris7t - please check if your copy of the `Dockerfile` has this line

Comment (kris7t) [reply Dockerfile:20]: Yeah, my `Dockerfile` had this line, but it didn't seem to have much effect. However, I tried moving it around a bit:

```diff
diff --git a/Dockerfile b/Dockerfile
index 52fc07b7..881332c1 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -11,14 +11,14 @@ WORKDIR /usr/src/ferdi
 COPY package*.json ./
 COPY lerna.json ./
 
+# Note: This is being set to bypass the error with missing git repo information for the 'preval-build-info' module
+ENV PREVAL_BUILD_INFO_PLACEHOLDERS=true
+
 RUN npm i node-gyp@8.0.0 node-sass@5.0.0 \
     && npx lerna bootstrap
 
 COPY . .
 
-# Note: This is being set to bypass the error with missing git repo information for the 'preval-build-info' module
-ENV PREVAL_BUILD_INFO_PLACEHOLDERS=true
-
 RUN cd recipes && npm i && npm run package && cd .. \
     && npm run build
```

Now I have no errors in my build: https://linx.marussy.com/s/uprjtznx.log Not even the config file one, which seems terribly strange (as the docker environment is supposed to be reproducible). For reference, the `node:14` image Docker uses as the base image for me has the image id `9153ee3e2ced`.

Comment (vraravam) [reply Dockerfile:20]: thanks for the catch @kris7t - very silly mistake.
But, as you said, it still doesn't explain how its different in yours vs mine. I am uploading this change now. Please approve the PR.

Comment (vraravam) [reply Dockerfile]: btw - the `node:14` docker image that I got has the same id: `9153ee3e2ced`

**Pull request merged**